#### 异常的语法：

1. try/except	`捕捉由python或你引起的异常并恢复。`
2. try/finally	`无论异常是否发生，执行清理行为`
3. raise		`手动在代码触发异常。`
4. assert		`有条件地在程序代码中触发异常。`
5. with/as		`在python2.6和后续版本中实现环境管理器。`

----


使用try捕捉异常
-
	def fetcher(obj,index):
		return obj[index]

	x = 'spam'
	try：
		fetcher(x,10)
	except IndexError:
		print 'got exception'

使用try/finally组合
- 
在finally里定义的语句不管是否发生了异常都会执行。

	try：
		fetcher(x,10)
	except IndexError:
		print 'got exception'
	finally:
		print 'fetch'
		
空的except会捕捉try中所有语句的异常
-

else
-
 在try中使用else会在没有异常时执行，在需要没有异常时才执行的语句时else很有用，如果在try/except执行后的语句虽然也会执行到，但是你不知道是发生了异常但是处理了还是没有发生异常执行到这块语句的。


	try：
		fetcher(x,10)
	except:
		print 'got except'
	else:
		print 'no except'

raise使用：故意触发异常
-
	raise <name>
	raise <name>,<data>
	raise

assert 语句
-
`assert <test>,<data>`,如果test为假则，python就会引发异常：data项（如果提供了的话）是异常的额外数据。